### ISSUE TITLE
Restoring from backup is not completely rolling back the application

### ISSUE THREAD

**Issue by twizzayy** [YunoHost/issues#1996]
# My YunoHost server

*Hardware:* Raspberry Pi 4B at home
*YunoHost version:* 4.3.6.3 (stable).
*I have access to my server:* Through SSH | through the webadmin | direct access via keyboard / screen

# Application that is experiencing the bug;
Pleroma

# Description

I tried to install a few new Front End from the Admin console recently. After reboot, I received nginx 502 errors.

Not A big deal. I had made a backup earlier that day, so I thought I would restore from that and all would be fixed.

After restoring from a previous local backup of my pleroma installation, the Known timeline is no longer loading and the home timeline is very slow and only loads occasionally. I am still visible to other instances. Can still post etc. but nothing from anyone else gets through the known timeline.

It should be noted that my current backup method is only a local backup of the pleroma installation. Using Yunohost's native backup method. Not a full disk backup. I noticed after restoring from the backup that the FE’s I had installed were still there. I deleted them, but there seems to be other changes that didn't get reverted because restoring from the backup is not solving the problem as a backup typically does.

Previous Backups and restorations have worked just fine. Not sure what went wrong with this one.

This Application worked as expected before I made these changes. I will shame myself and find better ways to back up my instance. But, I think it is worth investigating why restoring from a backup is failing to completely revert an application.

Would love to provide as much information as needed. I am very motivated to solve this problem. Very little experience with DB's and self-hosting. TIA

**Comment by Bugsbane** [comment]
From memory I had a similar issue to this quite a long time ago and was told that essentially the backup was just creating an archive, and the restore was trying to unpack the archive in the same place, but without removing the files already there.

It seemed to be assumed that the user would remove the app folder before restoring, which I felt was non-obvious. So my question would be whether anything was different other than extra files still existing, or files being the old versions after the restore?

**Comment by twizzayy** [comment]
After using pleroma for A while, I have found that the issue is indeed configuration folders persisting beyond installation.

I actually think this is a goof feature but, needs to be covered in a doc a bit more.
I am considering writing a readme for this repo that explains some of the things I have learned.

If you are experiencing this issue try removing files located at /etc/pleroma or /var/lib/pleroma that may have persisted beyond your backup.
